cpp/src/gandiva/CMakeLists.txt

```diff
@@ -28,7 +28,7 @@ add_dependencies(gandiva-all gandiva gandiva-tests gandiva-benchmarks)
 if(ARROW_DEPENDENCY_SOURCE STREQUAL "CONDA")
   set(LLVM_ROOT $ENV{CONDA_PREFIX})
 endif()
-find_package(LLVM)
+find_package(LLVM MODULE)
```

Comment:
Why was this added? So that `cpp/cmake_modules/FindLLVM.cmake` will be used even if `-DCMAKE_FIND_PACKAGE_PREFER_CONFIG=ON` is specified? In any case, please add a comment describing why this option is set

Reply:
That is correct. I'll add the comment.

Reply:
Please explicitly mention our custom `FindLLVM.cmake` and the fact that it will override `-DCMAKE_FIND_PACKAGE_PREFER_CONFIG=ON` (it's something of which users may not be aware, so this is a potentially surprising gotcha for the next person who uses that option)

Reply:
I extended the comment to provide a clearer reason why this is necessary here.

Reply:
Thanks!